File: src/dotnet/Ansight/HostConnectionResult.cs
namespace Ansight;

/// <summary>
/// Result returned by runtime-owned host connection actions.
/// </summary>
public sealed record HostConnectionResult(
    bool Success,
    string Message,
    HostConnectionActionKind Kind = HostConnectionActionKind.None,
    HostConnectionSource Source = HostConnectionSource.None,
    string? ReasonCode = null)
{
    public static HostConnectionResult FromSuccess(
        string message,
        HostConnectionActionKind kind = HostConnectionActionKind.None,
        HostConnectionSource source = HostConnectionSource.None,
        string? reasonCode = null)
        => new(true, message, kind, source, reasonCode);

    public static HostConnectionResult FromFailure(
        string message,
        HostConnectionActionKind kind = HostConnectionActionKind.None,
        HostConnectionSource source = HostConnectionSource.None,
        string? reasonCode = null)
        => new(false, message, kind, source, reasonCode);
}

File: src/ios/Sources/AnsightKit/HostConnectionResult.swift
import Foundation

public struct HostConnectionResult: Sendable, Codable, Equatable {
    public let success: Bool
    public let message: String
    public let kind: HostConnectionRequestKind
    public let source: HostConnectionSource
    public let reasonCode: String?
    public let openSession: OpenSessionResult?

    public init(
        success: Bool,
        message: String,
        kind: HostConnectionRequestKind,
        source: HostConnectionSource,
        reasonCode: String? = nil,
        openSession: OpenSessionResult? = nil
    ) {
        self.success = success
        self.message = message
        self.kind = kind
        self.source = source
        self.reasonCode = reasonCode
        self.openSession = openSession
    }
}
